BusinessLayer/JsonMessageBuilder/JsonMessageBuilder.cpp

```diff
@@ -0,0 +1,337 @@
+    m1ErrorFlagsJson[JsonFormat::SOFTWARE_OVER_CURRENT] = data.m1SoftwareOverCurrent();
+    m1ErrorFlagsJson[JsonFormat::DC_BUS_OVER_VOLTAGE] = data.m1DcBusOverVoltage();
+    m1ErrorFlagsJson[JsonFormat::BAD_MOTOR_POSITION_HALL_SEQUENCE] = data.m1BadMotorPositionHallSequence();
+    m1ErrorFlagsJson[JsonFormat::WATCHDOG_CAUSED_LAST_RESET] = data.m1WatchdogCausedLastReset();
+    m1ErrorFlagsJson[JsonFormat::CONFIG_READ_ERROR] = data.m1ConfigReadError();
+    m1ErrorFlagsJson[JsonFormat::RAIL_15V_UNDER_VOLTAGE_LOCK_OUT] = data.m1RailUnderVoltageLockOut();
+    m1ErrorFlagsJson[JsonFormat::DESATURATION_FAULT] = data.m1DesaturationFault();
+    motor1FaultsJson[JsonFormat::ERROR_FLAGS] = m1ErrorFlagsJson;
+
+    QJsonObject m0LimitsFlagsJson = QJsonObject();
+    m0LimitsFlagsJson[JsonFormat::OUTPUT_VOLTAGE_PWM] = data.m0OutputVoltagePwmLimit();
+    m0LimitsFlagsJson[JsonFormat::MOTOR_CURRENT] = data.m0MotorCurrentLimit();
+    m0LimitsFlagsJson[JsonFormat::VELOCITY] = data.m0VelocityLimit();
+    m0LimitsFlagsJson[JsonFormat::BUS_CURRENT] = data.m0BusCurrentLimit();
+    m0LimitsFlagsJson[JsonFormat::BUS_VOLTAGE_UPPER] = data.m0BusVoltageUpperLimit();
+    m0LimitsFlagsJson[JsonFormat::BUS_VOLTAGE_LOWER] = data.m0BusVoltageLowerLimit();
+    m0LimitsFlagsJson[JsonFormat::IPM_OR_MOTOR_TEMPERATURE] = data.m0IpmOrMotorTelemetryLimit();
+    motor0FaultsJson[JsonFormat::LIMITS_FLAGS] = m0LimitsFlagsJson;
+
+    QJsonObject m1LimitsFlagsJson = QJsonObject();
+    m1LimitsFlagsJson[JsonFormat::OUTPUT_VOLTAGE_PWM] = data.m1OutputVoltagePwmLimit();
+    m1LimitsFlagsJson[JsonFormat::MOTOR_CURRENT] = data.m1MotorCurrentLimit();
+    m1LimitsFlagsJson[JsonFormat::VELOCITY] = data.m1VelocityLimit();
+    m1LimitsFlagsJson[JsonFormat::BUS_CURRENT] = data.m1BusCurrentLimit();
+    m1LimitsFlagsJson[JsonFormat::BUS_VOLTAGE_UPPER] = data.m1BusVoltageUpperLimit();
+    m1LimitsFlagsJson[JsonFormat::BUS_VOLTAGE_LOWER] = data.m1BusVoltageLowerLimit();
+    m1LimitsFlagsJson[JsonFormat::IPM_OR_MOTOR_TEMPERATURE] = data.m1IpmOrMotorTelemetryLimit();
+    motor1FaultsJson[JsonFormat::LIMITS_FLAGS] = m1LimitsFlagsJson;
+
+    motor0FaultsJson[JsonFormat::RX_ERROR_COUNT] = data.getM0CanRxErrorCount();
+    motor0FaultsJson[JsonFormat::TX_ERROR_COUNT] = data.getM0CanTxErrorCount();
+    motor1FaultsJson[JsonFormat::RX_ERROR_COUNT] = data.getM1CanRxErrorCount();
+    motor1FaultsJson[JsonFormat::TX_ERROR_COUNT] = data.getM1CanTxErrorCount();
+    motorFaultsJson.append(motor0FaultsJson);
+    motorFaultsJson.append(motor1FaultsJson);
+
+    return motorFaultsJson;
+}
+
+QJsonArray JsonMessageBuilder::buildMpptMessage(const I_MpptData& data)
+{
+    QJsonArray mpptJson = QJsonArray();
+
+    data.getNumberOfUnits();
+
+    for (int i = 0; i < data.getNumberOfUnits(); i++)
+    {
+        QJsonObject mpptJsonUnit = QJsonObject();
+        const I_MpptUnit& mpptUnit = data.getMpptUnit(i);
+
+        if (mpptUnit.getMpptStatus())
```

Comment:
If in serial the mppt status message is the mppt number on bits 0-2 and the alive bit on bit 7, wouldn't this return true if the mppt number is there but the alive bit is not on?

Reply:
going up a level, this value isn't actually even set anywhere.

Reply:
Fixed it, the relevant changes were in `MpptMessage.cpp` and `MpptPopulator.cpp`